worker/apicaller/connect.go

```diff
@@ -141,12 +142,15 @@ func connectFallback(
 	// Yes, it's dumb that this can't be interrupted, and that
 	// it's not configurable without patching.
 	if params.IsCodeNotProvisioned(err) {
-		for a := checkProvisionedStrategy.Start(); a.Next(); {
+		retryStrategy := checkProvisionedStrategy
+		retryStrategy.Func = func() error {
 			tryConnect()
-			if !params.IsCodeNotProvisioned(err) {
-				break
+			if params.IsCodeNotProvisioned(err) {
```

Comment:
This isn't quite right. 

Have a look at the `IsFatalError` member of retry `CallArgs`. This should be used to inform that a `NotProvisioned` error is not fatal, but others are.

Then we want to assign the retry result to `err` on line 153.